src/utils/work/fetchWork.js

```diff
@@ -68,20 +68,28 @@ const fetchGeneExpressionWork = async (
   // Then, we may be able to find this in S3.
   let response = await seekFromS3(ETag, experimentId);
 
+  // If there is no response in S3, dispatch workRequest via the worker
+  let workResultReady = false;
   if (!response) {
-    response = await seekFromAPI(
+    workResultReady = await dispatchWorkRequest(
       experimentId,
       missingGenesBody,
       timeout,
       ETag,
-      null,
       {
         ETagPipelineRun: qcPipelineStartDate,
         ...extras,
       },
     );
   }
 
+  if (!workResultReady) {
+    console.debug(`No response immediately resolved for ${body} (ETag: ${ETag}) -- this is probably an event subscription.`);
```

Comment:
what does "this is probably an event subscription" mean?

Reply:
This error happens when there is a work response returned from the API which does not come from the result of a work request. One of them is the subscription of the UI to the socket channel. I don't think we need this anymore because workResultReady will be either `true`, or throw an error. I'll remove this.

Reply:
removed